services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/impl/kohsuke/GitHubServiceFactoryImpl.java

```diff
@@ -50,19 +47,23 @@ public GitHubService create(final String githubToken, final String githubUsernam
             final Cache cache = new Cache(githubCacheFolder, TENMB);
             final GitHubBuilder ghb = new GitHubBuilder()
                     .withConnector(new OkHttpConnector(new OkUrlFactory(new OkHttpClient().setCache(cache))));
-            if (githubUsername == null) {
-                ghb.withOAuthToken(githubToken);
-            } else {
-                ghb.withOAuthToken(githubToken, githubUsername);
-            }
+            identity.accept(new IdentityVisitor() {
+                @Override
+                public void visit(TokenIdentity token) {
+                    ghb.withOAuthToken(token.getToken());
+                }
+
+                @Override
+                public void visit(UserPasswordIdentity userPassword) {
+                    ghb.withPassword(userPassword.getUsername(), userPassword.getPassword());
+                }
+            });
             gitHub = ghb.build();
         } catch (final IOException ioe) {
             throw new RuntimeException("Could not create GitHub client", ioe);
         }
-        final GitHubService ghs = new KohsukeGitHubServiceImpl(gitHub, githubToken);
-        if (log.isLoggable(Level.FINEST)) {
-            log.log(Level.FINEST, "Created backing GitHub client for user " + githubUsername);
-        }
+        final GitHubService ghs = new KohsukeGitHubServiceImpl(gitHub, identity);
+        log.finest(() -> "Created backing GitHub client for identity " + identity);
```

Comment:
Let's future-proof it.  Someone may later change the `toString` impl and then we'll have a serious leak on our hands.  Better to explicitly call the `identityHashCode` if that's what we want..